pkg/datapath/linux/node.go

```diff
@@ -1150,6 +1150,9 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 // NodeValidateImplementation is called to validate the implementation of the
 // node in the datapath
 func (n *linuxNodeHandler) NodeValidateImplementation(nodeToValidate nodeTypes.Node) error {
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+
 	return n.nodeUpdate(nil, &nodeToValidate, false)
```

Comment:
nits: I saw in the codebase that normally we add a note on method with some details like `must be called with Lock() held`, do you think it will add any values if we do the same for _nodeUpdate_ as well as _nodeDelete_ ?

Reply:
Good point. Given that the CI already passed and in case there are no other requested change, I think this could be addressed in a follow-up PR. It will also make backporting this PR easier.